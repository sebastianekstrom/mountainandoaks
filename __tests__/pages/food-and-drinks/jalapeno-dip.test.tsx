import { render, screen } from "@testing-library/react";
import React from "react";

import JalapenoDip from "pages/food-and-drinks/jalapeno-dip";

describe("JalapenoDip page", () => {
  it("renders title correctly", () => {
    render(<JalapenoDip />);
    const title = screen.queryAllByText(/Jalapeño dip/i);
    expect(title).toBeDefined();
  });

  it("renders the ingress text", () => {
    render(<JalapenoDip />);
    const ingress = screen.queryAllByText(
      /This dip, with its zesty jalapeño and hint of garlic/i,
    );
    expect(ingress).toBeDefined();
  });

  it("renders the correct ingredients", () => {
    render(<JalapenoDip />);
    const ingredientList = screen.getByTestId("INGREDIENT_LIST");
    expect(ingredientList).toMatchSnapshot();
  });

  it("renders the method steps", () => {
    render(<JalapenoDip />);
    const step = screen.queryAllByText(
      /Start by prepping all the ingredients/i,
    );
    expect(step).toBeDefined();
  });
});
