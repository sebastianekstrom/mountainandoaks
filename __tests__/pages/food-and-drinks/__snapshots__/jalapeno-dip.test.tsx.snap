// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JalapenoDip page renders the correct ingredients 1`] = `
<section
  class="mb-8 lg:mb-0"
  data-testid="INGREDIENT_LIST"
>
  <div>
     
  </div>
  <div
    class="flex flex-col"
  >
    <ul>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Mayonnaise
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          100
          g
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Sour cream
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          40
          g
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Pickled jalape√±o
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          17
          g
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Baby spinach
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          13
          g
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Lime juice
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          6
          g
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Cilantro (optional)
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          5
          g
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Garlic
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          3
          g
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Sugar
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          1.5
          g
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Salt
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        />
      </li>
    </ul>
  </div>
</section>
`;
